### ISSUE TITLE
Bug/Improvement: Correct handling of message sending

### ISSUE THREAD

**Issue by martenrichter** [jupyterlite/pyodide-kernel#143]
Recently a lot of bug reports regarding jupyter lite widget interactivity being broken:
https://github.com/jupyterlite/pyodide-kernel/issues/133
https://github.com/jupyterlite/jupyterlite/issues/1465
https://github.com/jupyter-book/thebe/issues/771
https://github.com/jupyter/try-jupyter/issues/54

I have ran jupyter lite for two days in the debugger (see here my post about it here:
https://github.com/jupyterlite/pyodide-kernel/issues/133
).

And I am now relatively confident that the incomplete implementation of ``get_parent` in `kernel.py` and the js side:
https://github.com/jupyterlite/pyodide-kernel/blob/b70f099252670535db4745b5abab680300dc84eb/packages/pyodide-kernel/py/pyodide-kernel/pyodide_kernel/kernel.py#L29

is causing the trouble. As the control and shell channel parent is mixing through this simplification. 
In contrast to:
https://github.com/ipython/ipykernel/blob/188f39c509f1056c3784aad6954498f31de955a3/ipykernel/kernelbase.py#L622

So I think that a patch for this behavior is required. And this may fix the issues. Do you agree?

Are you open for PR? (But I must say, I am very experienced in C++ and js (and a bit in ts), but not in python, but I think the changes do not require much understanding of the language details, so it is possible that I will give up, if I fail). Also I am not so sure how I can test the changes on pyodide-kernel alone. So input is appreciated.

P.S: I hope it is ok to open a separate issue, I thought in this way it is a bit more organized.

**Comment by martenrichter** [comment]
Ok, I tried to implement it. But it was not the cause. Anyway, the wrong ordering of message with their kernel header is the cause. I also tried to reuse the parent headers from the initial execution attempt, but this did not work either  (xeus has always the same parent msg_id, so I gave it a try). I will continue....

**Comment by martenrichter** [comment]
Ok, next thing, if I go after reverting back to the commit before it still works.
If I clean all temp directories (e.g., node_modules) and build and install again, it does not work. 
If I change back to the reverted head in git build again, it works.
This is weird and may point to a specific combination of package versions???

**Comment by martenrichter** [comment]
Additional things:
1.) Use test code:
```
%pip install -q ipywidgets
from ipywidgets import interact

# Function to handle text input
def display_text(value=''):
    return f"Updated value: {value}"

# Create an interactive Text widget
interact(display_text, value='Type something...');
```
Hit two keys on the keyboard simultaneously (I used for example g and h next to each other on my German keyboard), I can provoke the problem with 90% chance.
2.) I am not sure, if the message sending is really atomic in worker.ts. So are all transactions hidden in setup immediately or is there a message sending beneath it, when passing the parent header to python? THis may be the cause beneath the problem.

**Comment by martenrichter** [comment]
Fixed by #148
